### ISSUE TITLE
No errors reported for use-before-definition with block-scoped declarations across files

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#6853]
See `tests/cases/compiler/constDeclarations-useBeforeDefinition2.ts`.

file1.ts:
```ts
c
```

file2.ts
```ts
const c = 0;
```

Compile with `tsc --outFile hello.js file1.ts file2.ts`

Expected: Use before declaration error.
Actual: No error.

Pinging @vladima and @zhengbli for bringing this up.

**Comment by ahejlsberg** [comment]
Since we don't know anything about the final ordering unless you compile with `-outFile`, are you proposing we only do this with `-outFile`? Seems a bit odd.

**Comment by mhegazy** [comment]
@ahejlsberg, this is what we do, or were doing, anyways. see https://github.com/Microsoft/TypeScript/blob/master/src/compiler/checker.ts#L487-L493

**Comment by vladima** [comment]
I can repro the issue only if I specify `-t ES6` (the same setting is set in the test from the issue description). Root cause: we skip this check for modules because we cannot reason about ordering. However in current codebase having target `ES6`  also implies `moduleKind=ES6` and this causes us to bypass the check.

**Comment by mhegazy** [comment]
@zhengbli has a change to this logic already, to change module default. so i guess this should be covered.

**Comment by vladima** [comment]
@zhengbli can you test if this issue still reproduces with your change?

**Comment by zhengbli** [comment]
@vladima @mhegazy this issue can be fixed by changing the default module. I brought it up to make sure that this is the desired behavior. I'll send a PR soon.
